src/Casting/CarbonCast.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace WendellAdriel\ValidatedDTO\Casting;
+
+use Carbon\Carbon;
+use Throwable;
+use WendellAdriel\ValidatedDTO\Exceptions\CastException;
+
+class CarbonCast implements Castable
+{
+    /**
+     * @param  string|null  $timezone
+     */
+    public function __construct(private ?string $timezone = null)
```

Comment:
I think that on optional `$format` or even `$formats` would help here to enforce a given date format. As the auto guessing can be wrong sometimes.